test/ft/exceptions.cpp

```diff
@@ -239,14 +239,14 @@ test exception_and_context = [] {
       // clang-format off
         return make_transition_table(
            *idle + event<e1> / [] { throw exception1{}; } = s1
-          , s1 + exception<exception1> / [](const context &,const auto &){} = X
+          , s1 + exception<exception1> / [](context &,const auto &){} = X
         );
       // clang-format on
     }
   };
 
   context ctx{};
-  sml::sm<c> sm{&ctx};
+  sml::sm<c> sm{ctx};
```

Comment:
Oh... I was blind. :-)

Originally, you injected a pointer to the `context` object and in your action lambda you tried to retrieve it as a (`const`) reference to the object. That could never work and still will not work even with #140 being merged.

Your current changes, however, do work.

And together with #140 you can even reference `context` by `const` in the action lambda's argument-list.

Reply:
As I am not the author of Boost.SML it is not up to me to decide, but I would recommend
1. you wait until #140 is merged,
2. then rebase your MR on the latest `master` branch and squash your two commits into a single one and
3. additionally add `const` back to the action's argument again (in the same commit).

Thereby the history is clean and it does not contain any (new) errors (or bad practice) in its test-cases.

Reply:
Great, anything else we want to add before merging?

Reply:
Nothing I know of which belongs into this PR.
The only thing might be some more unit-tests that probably test some rare cases which also depend on your changes. But which...

Maybe copy your unit-test and change the expected argument-type of the action? (const reference, non-const reference, copy, pointer, const pointer, reference to pointer... Although most of the pointer-test-cases will probably not work.)

Reply:
As #140 is already merged you can proceed as suggested.